executor/include/execution/executor/base_executor.hpp

```diff
@@ -4,22 +4,33 @@
 
 #pragma once
 
+#include <array>
+#include <execution/property.hpp>
+#include <execution/type_trait.hpp>
 #include <functional>
 #include <iostream>
+#include <memory>
 #include <stdexcept>
 #include <string>
-#include <array>
-#include <memory>
-
-#include <execution/property.hpp>
-#include <execution/type_trait.hpp>
-
 
-template <template <typename, typename, typename, typename> typename Derived,
-    typename Interface, typename Cardinality, typename Blocking,
-    typename ProtoAllocator>
+template <template <typename...> typename Derived, typename Interface,
+          typename Cardinality, typename Blocking, typename ProtoAllocator>
 class executor {
  public:
+  template <typename Executor,
+            typename execution::instance_of_base<Derived, Executor> = 0>
+  bool operator==(const Executor &) const noexcept {
+    return std::is_same_v<
+        Derived<Interface, Cardinality, Blocking, ProtoAllocator>, Executor>;
+  }
```

Comment:
Binary and symmetric operators (sum, mult, equivalence, etc...) are usually recommended to bedefined outside the class.

Reply:
Any reason for this?

Reply:
http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Ro-symmetric
https://www.modernescpp.com/index.php/c-core-guidelines-rules-for-overloading-and-overload-operators
In your particular case the most pressing reason is because it  removes the need of passing the derived class as a template parameter to your base type.